src/components/tl/ucp/tl_ucp_team.c

```diff
@@ -96,6 +96,12 @@ ucc_status_t ucc_tl_ucp_team_create_test(ucc_base_team_t *tl_team)
     ucc_tl_ucp_context_t *ctx  = UCC_TL_UCP_TEAM_CTX(team);
     ucc_status_t          status;
 
+    if (USE_SERVICE_WORKER(team)) {
+        team->worker = &ctx->service_worker;
```

Comment:
does this line mean that we will use service_worker for all the subsequent communications or I'm missing something?

Reply:
@samnordmann yes, caching worker this way does not work. we have to use different workers for service colls and all the rest. so looks like we need the implementation with the branch - the one you originally madet.

Reply:
> does this line mean that we will use service_worker for all the subsequent communications or I'm missing something?
Yes.
I thought this is what we wanted, but ok I think I understand the problem now.



> @samnordmann yes, caching worker this way does not work. we have to use different workers for service colls and all the rest. so looks like we need the implementation with the branch - the one you originally made.

I see, so to be clear: adding a flag `is_service` at the task level and passing the flag to the relevant functions, right ?

Reply:
or we can cache "worker" on "task". Like you do now, but on "per-task" basis. Then you take branch once during task init.

Reply:
you are right. lets have a flag then. @Sergei-Lebedev ?

Reply:
> does this line mean that we will use service_worker for all the subsequent communications or I'm missing something?

I am actually not sure to understand the issue you raised in the first place @Sergei-Lebedev.
This means that we will use service_worker for all subsequent communications **in the service team**. The other teams will use the regular worker.
This is what we want no ? The service team is a special team only used for service collectives, right ?
If yes then there is nothing to fix.

Reply:
yes, missed it, thanks @samnordmann

Reply:
@samnordmann yes, you are right. Plz fix 1 liner error:
```
/home/runner/work/ucc/ucc/src/components/tl/ucp/tl_ucp_context.c:362:68: error: parameter 'ctx' is unused [misc-unused-parameters,-warnings-as-errors]
                                             ucc_tl_ucp_context_t *ctx)
```
by adding "//NOLINT and lets merge this

Reply:
The warning is actually relevant. I removed ctx from the arguments.